File: docs/examples/go/list_ping_services.go
// Command list_ping_services lists ping checks configured for a project.
package main

import (
    "encoding/json"
    "fmt"
    "net/http"
)

const (
    pingListToken = "<token>"
    pingListProjectUUID = "<project-uuid>"
)

func main() {
    req, _ := http.NewRequest(http.MethodGet, baseURL+"/projects/"+pingListProjectUUID+"/ping-services", nil)
    req.Header.Set("Authorization", "Bearer "+pingListToken)

    resp, err := http.DefaultClient.Do(req)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()

    if resp.StatusCode != http.StatusOK {
        panic(fmt.Sprintf("unexpected status: %d", resp.StatusCode))
    }

    var services []struct {
        Name     string `json:"name"`
        URL      string `json:"url"`
        Interval int    `json:"interval"`
    }
    if err := json.NewDecoder(resp.Body).Decode(&services); err != nil {
        panic(err)
    }

    for _, service := range services {
        fmt.Printf("%s -> %s (interval %ds)\n", service.Name, service.URL, service.Interval)
    }
}

File: docs/examples/python/list_ping_services.py
"""List ping services associated with a project."""

import requests

BASE_URL = "http://localhost:3000/api"
TOKEN = "<token>"
PROJECT_UUID = "<project-uuid>"


def list_ping_services() -> None:
    response = requests.get(
        f"{BASE_URL}/projects/{PROJECT_UUID}/ping-services",
        headers={"Authorization": f"Bearer {TOKEN}"},
        timeout=10,
    )
    response.raise_for_status()
    for service in response.json():
        print(f"{service['name']} -> {service['url']} (interval {service['interval']}s)")


if __name__ == "__main__":
    list_ping_services()
